sentry_sdk/scope.py

```diff
@@ -1251,3 +1452,69 @@ def __repr__(self):
             hex(id(self)),
             self._name,
         )
+
+
+def _with_new_scope():
+    # type: () -> Generator[Scope, None, None]
+
+    current_scope = Scope.get_current_scope()
+    forked_scope = current_scope.fork()
+    token = _current_scope.set(forked_scope)
+
+    try:
+        yield forked_scope
+
+    finally:
+        # restore original scope
+        _current_scope.reset(token)
+
+
+@contextmanager
+def new_scope():
+    # type: () -> Generator[Scope, None, None]
+    """
+    Context manager that forks the current scope and runs the wrapped code in it.
+
+    .. versionadded:: 1.XX.0
+    """
+    ctx = copy_context()  # This does not exist in Python 2.7
+    return ctx.run(_with_new_scope)
+
+
+def _with_isolated_scope():
+    # type: () -> Generator[Scope, None, None]
+
+    # fork current scope
+    current_scope = Scope.get_current_scope()
+    forked_current_scope = current_scope.fork()
+    current_token = _current_scope.set(forked_current_scope)
+
+    # fork isolation scope
+    isolation_scope = Scope.get_isolation_scope()
+    forked_isolation_scope = isolation_scope.fork()
+    isolation_token = _isolation_scope.set(forked_isolation_scope)
+
+    try:
+        yield forked_isolation_scope
+
+    finally:
+        # restore original scopes
+        _current_scope.reset(current_token)
+        _isolation_scope.reset(isolation_token)
+
+
+@contextmanager
+def isolated_scope():
```

Comment:
Wondering if it won't be confusing to have both `isolation_scope` and `isolated_scope` around in function/ctx manager names. Don't really have a better idea though, but leaving this here in case someone else does.